app/Models/TalkModel.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace App\Models;
+
+use CodeIgniter\Model;
+
+class TalkModel extends Model
+{
+    protected $table = 'Talk';
+    protected $allowedFields = ['event_id', 'speaker_id', 'starts_at', 'duration', 'title', 'description'];
+    protected $useTimestamps = true;
+
+    public function getByEventId(int $eventId): array
+    {
+        // get all talks for the given event
+        $talks = $this->db->table('Talk')
+            ->select('id, speaker_id, starts_at, duration, title, description')
+            ->where('event_id', $eventId)
+            ->orderBy('starts_at', 'ASC')
+            ->get()
+            ->getResultArray();
+
+        // get all tags
+        $tags = $this->db->table('TalkHasTag')
+            ->join('Tag', 'Tag.id = TalkHasTag.tag_id')
+            ->select('TalkHasTag.talk_id, Tag.text, Tag.color_index')
+            ->get()
+            ->getResultArray();
+
+        // add tags to the talks
+        foreach ($talks as &$talk) {
+            $tagsForThisTalk = array_filter($tags, function ($tag) use ($talk) {
+                return $tag['talk_id'] === $talk['id'];
+            });
```

Comment:
Is there a way to move the filter into the database?
Im not the best with databases but maybe there is a way to use a sub query to let the database select the tags and provide them already into the talk.
In this case you could also remove the extra tags database call.

Reply:
If I add it to the database, then I would get one entry per tag of each talk. Something like this:

| talk id | title | ... | tag_name | tag_color_index |
| ------ | -- | --- | --------- | ------------- |
| 1 | Katzen würden VISCA senden | ... | Programmierung | 1 |
| 1 | Katzen würden VISCA senden | ... | Maker | 2 |
| 1 | Katzen würden VISCA senden | ... | Reverse Engineering | 3 |
| 2 | Journey eines Rollenspiels | ... | Programmierung | 1 |
| 2 | Journey eines Rollenspiels  | ... | GameDev | 4 |
| ... | ... | ... | ... | ... |

In PHP, I would then have to "convert" that table into the JSON structure that we need. Since my chat told me that there's no problem if we do "a few database queries" for a single page load, I solved it this way. I hope that's okay 🤔